src/concurrency/thread.rs

```diff
@@ -905,7 +906,7 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
                     interp_ok(MiriAllocBytes::from_bytes(
                         std::borrow::Cow::Borrowed(bytes),
                         align,
-                        (),
+                        params.clone(),
```

Comment:
Ah damn, this closure should be a `FnOnce` and not `FnMut` and then we can avoid a clone here...
Could you make a rustc PR for that?

Reply:
Once https://github.com/rust-lang/miri/pull/4355 lands you should be able to rebase & remove the `clone`.